### PR TITLE
Build changes: esbuild & TypeScript

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Aim is to allow the majority of the JavaScript to move to TypeScript and
use multiple files as required while continuing to provide it as a
single JavaScript file for emscripten.

As a bonus we can clean up the SVG loading by bundling it, removing
async awkwardness at initialization time.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,13 +6,16 @@ all: build dist
...
 	$(MAKE) -C src
 
-dist:
+dist: build
 	mkdir -p $(DIST)/build
 	cp -r $(SRC)/*.html $(SRC)/*.svg $(SRC)/term.js src/examples $(DIST)
 	cp $(SRC)/build/micropython.js $(SRC)/build/firmware.wasm  $(DIST)/build/
 
+watch: dist
+	fswatch -o -e src/build src  | while read _; do $(MAKE) dist; done
```
Comment (microbit-matt-hillsdon) [inline Makefile:15]: This is a bit of a hack but I've been using it for a while and it's a big workflow improvement for me.

https://github.com/emcrisostomo/fswatch

File: Makefile [hunk 2]
```diff
@@ -6,13 +6,16 @@ all: build dist
 build:
 	$(MAKE) -C src
 
-dist:
+dist: build
 	mkdir -p $(DIST)/build
 	cp -r $(SRC)/*.html $(SRC)/*.svg $(SRC)/term.js src/examples $(DIST)
```
Comment (microbit-matt-hillsdon) [inline Makefile]: SVG is bundled so no need to copy it to dist.

```suggestion
	cp -r $(SRC)/*.html $(SRC)/term.js src/examples $(DIST)
```

**Review by microbit-robert — APPROVED** [review]
LGTM.

**Comment by microbit-matt-hillsdon** [comment]
I think this is a good start. It might be worth revisiting soon to add a test runner. I'll get this deploying to a microbit.org domain first.

**Pull request merged**